lib/client.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+require 'redis'
+
+class SentinelClient
+  def initialize(master_name, credentials)
+    @client = Redis.new(
+      host: credentials.fetch("host"),
+      port: credentials.fetch("port"),
+    )
+    @master_name = master_name
+  end
+
+  def get_master_info
+    @client.sentinel("master", @master_name)
+  end
+
+  def get_replicas_info
+    @client.sentinel("replicas", @master_name)
+  end
+
+  def fail_over
+    @client.sentinel("failover", @master_name)
+  end
+end
+
+class RedisClient
+  def initialize
+  end
+
+  def self.default(credentials)
+    Redis.new(
+      host: credentials.fetch('host'),
+      port: credentials.fetch('port'),
+      password: credentials.fetch('password'),
+      timeout: 30
+    )
+  end
+  def self.using_sentinel(credentials)
+    Redis.new(
+      host: credentials.fetch('master_name'),
+      password: credentials.fetch('password'),
+      sentinels: credentials.fetch('sentinels').map { | sentinel |  { host: sentinel["host"], port: sentinel["port"] } },
+      timeout: 30
+    )
+  end
+
+  def self.tls(credentials, version='')
+    ssl_params = {
+      verify_mode: OpenSSL::SSL::VERIFY_NONE,
+      version: version,
```

Comment:
version will not work with ssl_params instead ssl_version will work.

Reply:
done